platform/branding-registry-impl/src/main/java/org/codice/ddf/branding/impl/BrandingRegistryImpl.java

```diff
@@ -65,11 +104,42 @@ public String getProductVersion() {
 
   @Override
   public List<BrandingPlugin> getBrandingPlugins() {
-    return brandingPlugins;
+    if (brandingPlugins.isEmpty()) {
```

Comment:
I think there's an issue with this implementation. Say
1. BrandingPlugin1 starts
2. BrandingRegistryImpl starts
3. BrandingPlugin2 starts

BrandingRegistryImpl will be notified when BrandingPlugin2 is registered, but not BrandingPlugin1. When getBrandingPlugins() is called, it will return a list with only BrandingPlugin2.

You should be able to fix this by always performing this logic when you're adding the service listener. Then the brandingPlugin list will reflect any services that were started before the registry and the service listener will take care of adding any that are registered after

Reply:
It wouldn't be notified, which is why I lazy initialize the list if its empty. This does work, by the way, with the scenario that you mentioned, it is exactly what happens for a downstream.

Reply:
It works because this is only an issue when going through the installer, which immediately calls this method.

Reply:
What I'm suggesting is that the list wouldn't be empty if BrandingPlugin2 is registered, and so BrandingPlugin1 would never be added. 

If it works, I'm curious how

Reply:
Ah, ok. That makes sense. But we're making assumptions then about how this class will be used. Imo it would be better if it worked in the general case